src/extension.ts

```diff
@@ -847,6 +849,28 @@ function createMonitor(): any {
     });
 }
 
+function createIdfTerminal() {
+    PreCheck.perform([webIdeCheck, openFolderCheck], () => {
+        const envSetCmd = process.platform === "win32" ? "set" : "export";
+        const modifiedEnv = utils.appendIdfAndToolsToPath();
```

Comment:
also, question, why are you adding esptool, espcoredump, partition_table paths here, and not in appendIdfAndToolsToPath function? Looks like that might be easier to keep all extra paths in one place, next to `path.join(modifiedEnv.IDF_PATH, "tools")`.